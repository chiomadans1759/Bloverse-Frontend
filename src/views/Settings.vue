<template>
<main id="settings">
  <div class="container">
  <div class="row d-flex justify-content-center align-items-center">
  <div class="col-sm-7 col-lg-6">
    <h3 class="pb-4">Settings</h3>
    <div class="card p-4">
      <h6 class="pb-3">Profile</h6>

      <a href="#" class="text-capitalize">edit profile</a>
      <p class="text small">Update your information</p>

      <a href="#" class="text-capitalize mt-3">change password</a>
      <p class="text small">Update your information</p>

      <a href="#" class="text-capitalize mt-3">customize country</a>
      <p class="text small">Update your information</p>

      <a href="#" class="text-capitalize mt-3">customize category</a>
      <p class="text small">Update your information</p>

      <a href="#" class="text-capitalize mt-3">feedback</a>
      <p class="text small">Update your information</p>   

      <p class="text-capitalize font-weight-bold pt-4 auth">authorization</p> 

        <div class="pt-3 d-flex justify-content-between align-items-center">
          <div class="pr-2">
            <p class="text-capitalize">newsletter subscription</p>
            <p class="text small"> Subscribe to receive article digest of your interest</p>
          </div>   
           <div >
            <button type="button" class="btn btn-sm btn-toggle active" data-toggle="button" aria-pressed="true" autocomplete="off">
              <div class="square"></div>
            </button>
          </div>
        </div> 

        <div class="pt-3 d-flex justify-content-between align-items-center">
          <div>
            <p>Browser Push Notifications</p>
            <p class="text small"> Receive article digest</p>
          </div>   
           <div>
            <button type="button" class="btn btn-sm btn-toggle active" data-toggle="button" aria-pressed="true" autocomplete="off">
              <div class="square"></div>
            </button>
          </div>
        </div>  

        <div class="hr"></div>

        <div class="d-flex justify-content-between align-items-center">
          <div>
            <p class="text-capitalize">Become a Content Creator</p>
            <p class="text small mb-2">Publish your article to the world</p>
            <a href="" class="text-capitalize">know more</a>
          </div>   
           <div>
            <button type="button" class="btn btn-sm btn-primary text-capitalize">apply</button>
          </div>

        </div> 
    </div>
  </div>
  </div>
  </div>
</main>
  
</template>

<script>
export default {
  
}
</script>

<style scoped>

  #settings {
    width: 100%;
    margin-bottom: 7rem;
  }

  #settings h1 {
   margin-left: 6rem;
  }

  #settings .card {
    background: #fff;
    height: auto;
    padding-top: 1.5rem;
    padding-bottom: 7rem;
  }

  #settings .card .text {
    color: #868686;
  }

  #settings .card .auth {
    font-size: 0.9rem;
  }

 #settings .btn-toggle{
  padding: 0;
  position: relative;
  border: none;
  height: 1.5rem;
  width: 3rem;
  background: rgb(131, 128, 128);
  border-radius: 1.5rem;
}
#settings .btn-toggle.btn-sm:focus,
.btn-toggle.btn-sm.focus,
.btn-toggle.btn-sm:focus.active,
.btn-toggle.btn-sm.focus.active {
  outline: none;
}
#settings .btn-toggle.btn-sm:before,
.btn-toggle.btn-sm:after {
  line-height: 1.5rem;
  width: 0.5rem;
  text-align: center;
  font-weight: 600;
  font-size: 0.55rem;
  text-transform: uppercase;
  letter-spacing: 2px;
  position: absolute;
  bottom: 0;
  transition: opacity 0.25s;
}
#settings .btn-toggle.btn-sm:before {
  content: 'Off';
  left: -0.5rem;
}
#settings .btn-toggle.btn-sm:after {
  content: 'On';
  right: -0.5rem;
  opacity: 0.5;
}
#settings .btn-toggle.btn-sm > .square {
  position: absolute;
  top: 0.1875rem;
  left: 0.1875rem;
  width: 1.125rem;
  height: 1.125rem;
  background: #fff;
  transition: left 0.25s;
}
#settings .btn-toggle.btn-sm.active {
  transition: 0.25s;
  background: #096DD9;
}
#settings .btn-toggle.btn-sm.active > .square {
  left: 1.6875rem;
  transition: left 0.25s;
}
#settings .btn-toggle.btn-sm.active:before {
  opacity: 0.5;
}
#settings .btn-toggle.btn-sm.active:after {
  opacity: 1;
}
#settings .btn-toggle.btn-sm.btn-sm:before,
.btn-toggle.btn-sm.btn-sm:after {
  line-height: -0.5rem;
  color: #fff;
  letter-spacing: 0.75px;
  left: 0.4125rem;
  width: 2.325rem;
}
#settings .btn-toggle.btn-sm.btn-sm:before {
  text-align: right;
}
#settings .btn-toggle.btn-sm.btn-sm:after {
  text-align: left;
  opacity: 0;
}
#settings .btn-toggle.btn-sm.btn-sm.active:before {
  opacity: 0;
}
#settings .btn-toggle.btn-sm.btn-sm.active:after {
  opacity: 1;
}

#settings .hr {
  margin-top: 1rem;
  margin-bottom: 1rem;
  border: 0;
  border-top: 1px solid rgba(0, 0, 0, 0.10);
}

</style>
