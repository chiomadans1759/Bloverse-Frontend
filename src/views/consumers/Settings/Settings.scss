#settings {
  width: 100%;
  margin-bottom: 7rem;
  h1 {
    margin-left: 6rem;
  }
  .card {
    background: #ffffff;
    height: auto;
    padding-top: 1.5rem;
    padding-bottom: 7rem;
    .text {
      color: #868686;
    }
    .auth {
      font-size: 0.9rem;
    }
  }
  .btn-toggle {
    padding: 0;
    position: relative;
    border: none;
    height: 1.5rem;
    width: 3rem;
    background: rgb(131, 128, 128);
    border-radius: 1.5rem;
    &.btn-sm:focus {
      outline: none;
    }
  }
}

.btn-toggle.btn-sm {
  &.focus, &:focus.active, &.focus.active {
    outline: none;
  }
}

#settings .btn-toggle.btn-sm:before, .btn-toggle.btn-sm:after {
  line-height: 1.5rem;
  width: 0.5rem;
  text-align: center;
  font-weight: 600;
  font-size: 0.55rem;
  text-transform: uppercase;
  letter-spacing: 0.1rem;
  position: absolute;
  bottom: 0;
  transition: opacity 0.25s;
}

#settings .btn-toggle.btn-sm {
  &:before {
    content: 'Off';
    left: -0.5rem;
  }
  &:after {
    content: 'On';
    right: -0.5rem;
    opacity: 0.5;
  }
  > .square {
    position: absolute;
    top: 0.1875rem;
    left: 0.1875rem;
    width: 1.125rem;
    height: 1.125rem;
    background: #ffffff;
    transition: left 0.25s;
  }
  &.active {
    transition: 0.25s;
    background: #096DD9;
    > .square {
      left: 1.6875rem;
      transition: left 0.25s;
    }
    &:before {
      opacity: 0.5;
    }
    &:after {
      opacity: 1;
    }
  }
  &.btn-sm:before {
    line-height: -0.5rem;
    color: #ffffff;
    letter-spacing: 0.1rem;
    left: 0.4125rem;
    width: 2.325rem;
  }
}

.btn-toggle.btn-sm.btn-sm:after {
  line-height: -0.5rem;
  color: #ffffff;
  letter-spacing: 0.1rem;
  left: 0.4125rem;
  width: 2.325rem;
}

#settings {
  .btn-toggle.btn-sm.btn-sm {
    &:before {
      text-align: right;
    }
    &:after {
      text-align: left;
      opacity: 0;
    }
    &.active {
      &:before {
        opacity: 0;
      }
      &:after {
        opacity: 1;
      }
    }
  }
  .hr {
    margin-top: 1rem;
    margin-bottom: 1rem;
    border: 0;
    border-top: 0.09rem solid rgba(0, 0, 0, 0.1);
  }
}