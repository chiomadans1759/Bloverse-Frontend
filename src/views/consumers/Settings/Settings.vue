<template>
<main id="settings">
  <div class="container">
  <div class="row d-flex justify-content-center align-items-center">
  <div class="col-sm-7 col-lg-6">
    <h3 class="pb-4">Settings</h3>
    <div class="card p-4">
      <h6 class="pb-3">Profile</h6>

      <a href="#" class="text-capitalize">edit profile</a>
      <p class="text small">Update your information</p>

      <a href="#" class="text-capitalize mt-3">change password</a>
      <p class="text small">Update your information</p>

      <a href="#" class="text-capitalize mt-3">customize country</a>
      <p class="text small">Update your information</p>

      <a href="#" class="text-capitalize mt-3">customize category</a>
      <p class="text small">Update your information</p>

      <a href="#" class="text-capitalize mt-3">feedback</a>
      <p class="text small">Update your information</p>   

      <p class="text-capitalize font-weight-bold pt-4 auth">authorization</p> 

        <div class="pt-3 d-flex justify-content-between align-items-center">
          <div class="pr-2">
            <p class="text-capitalize">newsletter subscription</p>
            <p class="text small"> Subscribe to receive article digest of your interest</p>
          </div>   
           <div >
            <button type="button" class="btn btn-sm btn-toggle active" data-toggle="button" aria-pressed="true" autocomplete="off">
              <div class="square"></div>
            </button>
          </div>
        </div> 

        <div class="pt-3 d-flex justify-content-between align-items-center">
          <div>
            <p>Browser Push Notifications</p>
            <p class="text small"> Receive article digest</p>
          </div>   
           <div>
            <button type="button" class="btn btn-sm btn-toggle active" data-toggle="button" aria-pressed="true" autocomplete="off">
              <div class="square"></div>
            </button>
          </div>
        </div>  

        <div class="hr"></div>

        <div class="d-flex justify-content-between align-items-center">
          <div>
            <p class="text-capitalize">Become a Content Creator</p>
            <p class="text small mb-2">Publish your article to the world</p>
            <a href="" class="text-capitalize">know more</a>
          </div>   
           <div>
            <button type="button" class="btn btn-sm btn-primary text-capitalize">apply</button>
          </div>

        </div> 
    </div>
  </div>
  </div>
  </div>
</main>
  
</template>

<script>
export default {
  
}
</script>

<style lang="scss" scoped>
  @import './Settings';
</style>

