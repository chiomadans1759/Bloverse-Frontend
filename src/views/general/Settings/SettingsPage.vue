<template>
  <main id="settings">
    <div class="container">
      <div class="row d-flex justify-content-center align-items-center">
        <div class="col-sm-7 col-lg-6">
          <h3 class="pb-4">Settings</h3>
          <div class="card p-4">
            <h6 class="pb-3">Profile</h6>

            <router-link to="/edit/profile" class="text-capitalize">edit profile</router-link>
            <p class="text small">Update your information</p>

            <router-link to="/changepassword" class="text-capitalize mt-3">change password</router-link>
            <p class="text small">Update your information</p>

            <router-link to="/choosecountry" class="text-capitalize mt-3">customize country</router-link>
            <p class="text small">Update your information</p>

            <router-link to="/choosecategory" class="text-capitalize mt-3">customize category</router-link>
            <p class="text small">Update your information</p>

            <a data-toggle="modal" href="#feedback-modal" class="text-capitalize mt-3" v-if="!isCreator">give feedback</a>
            <p class="text small" v-if="!isCreator">Update your information</p>

            <router-link to="/creators/social-media" class="text-capitalize mt-3" v-if="!isCreator">Update Social Media</router-link>
            <p class="text small" v-if="!isCreator">Update your social media information</p>

            <p class="text-capitalize font-weight-bold pt-4 auth">authorization</p>

            <div class="pt-3 d-flex justify-content-between align-items-center" v-if="!isCreator">
              <div class="pr-2">
                <p class="text-capitalize">newsletter subscription</p>
                <p class="text small">Subscribe to receive article digest of your interest</p>
              </div>
              <div>
                <button
                  type="button"
                  class="btn btn-sm btn-toggle active"
                  data-toggle="button"
                  aria-pressed="true"
                  autocomplete="off"
                >
                  <div class="square"></div>
                </button>
              </div>
            </div>

            <div class="pt-3 d-flex justify-content-between align-items-center" v-if="isCreator">
              <div class="pr-2">
               <router-link to="/creators/email-settings" class="text-capitalize">Manage Email Preferences</router-link>
                <p class="text small">Customize email notifications</p>
              </div>
            </div>

            <div class="pt-3 d-flex justify-content-between align-items-center" v-if="isCreator">
              <div class="pr-2">
                <p class="text-capitalize">switch to consumer account</p>
                <p class="text small">switch to consumer view</p>
              </div>
              <div>
                <button
                  type="button"
                  class="btn btn-sm btn-toggle active"
                  data-toggle="button"
                  aria-pressed="true"
                  autocomplete="off"
                >
                  <div class="square"></div>
                </button>
              </div>
            </div>

            <div class="pt-3 d-flex justify-content-between align-items-center">
              <div>
                <p>Browser Push Notifications</p>
                <p class="text small">All notifications</p>
              </div>
              <div>
                <button
                  type="button"
                  class="btn btn-sm btn-toggle active"
                  data-toggle="button"
                  aria-pressed="true"
                  autocomplete="off"
                >
                  <div class="square"></div>
                </button>
              </div>
            </div>

            <div class="pt-3 d-flex justify-content-between align-items-center" v-if="isCreator">
              <div>
                <p>Share Posts on Social Media</p>
                <p class="text small">Auto-share to all social media platforms</p>
              </div>
              <div>
                <button
                  type="button"
                  class="btn btn-sm btn-toggle active"
                  data-toggle="button"
                  aria-pressed="true"
                  autocomplete="off"
                >
                  <div class="square"></div>
                </button>
              </div>
            </div>

            <div class="hr"></div>

            <div class="d-flex justify-content-between align-items-center" v-if="isCreator">
              <div>
                <p class="text-capitalize">Become a Content Consumer</p>
                <p class="text small mb-2">Publish your article to the world</p>
              </div>
              <div>
                <router-link to="#" class="btn btn-sm btn-primary text-capitalize">get started</router-link>
              </div>
            </div>
            <div class="d-flex justify-content-between align-items-center" v-if="!isCreator">
              <div>
                <p class="text-capitalize">Become a Content Creator</p>
                <p class="text small mb-2">Publish your article to the world</p>
                <router-link to="/creators" class="text-capitalize">know more</router-link>
              </div>
              <div>
                <router-link
                  to="/creators/apply"
                  class="btn btn-sm btn-primary text-capitalize"
                >apply</router-link>
              </div>
            </div>
          </div>
        </div>
      </div>
      <modal
        target="feedback-modal"
        altTitle="Send Us Some Feedback"
        actionText="Send Feedback"
        dark
        subTitle="Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!"
      >
        <div class="bl-modal-content">
          <p>Enter your feedback here!</p>
          <textarea class="form-control mb-2 p-1"></textarea>
          <label class="ml-1 mr-3">
            <input type="radio" name="feedback-type"> Bug
          </label>
          <label class="mr-3">
            <input type="radio" name="feedback-type"> Comment
          </label>
          <label class="mr-3">
            <input type="radio" name="feedback-type"> Other
          </label>
        </div>
      </modal>
    </div>
  </main>
</template>

<script>
import Modal from "@/components/Modal/Modal";

export default {
  components: { Modal },
  props: {
    isCreator: {
      type: Boolean,
      default: true
    }
  }
};
</script>

<style lang="scss" scoped>
@import "./SettingsPage";
</style>

