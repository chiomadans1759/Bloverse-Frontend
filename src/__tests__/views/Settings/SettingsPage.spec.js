import { mount, RouterLinkStub  } from '@vue/test-utils';
import Settings from '@/views/consumers/Settings/SettingsPage.vue';

describe('Settings Page', () => {
  it('should render correctly', () => {

    const wrapper = mount(Settings, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.find(RouterLinkStub).props().to).toBe('/edit/profile')
    expect(wrapper.element).toMatchSnapshot();
  });
});
