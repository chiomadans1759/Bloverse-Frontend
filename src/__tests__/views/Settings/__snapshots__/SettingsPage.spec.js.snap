// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Page should render correctly 1`] = `
<main
  id="settings"
>
  <div
    class="container"
  >
    <div
      class="row d-flex justify-content-center align-items-center"
    >
      <div
        class="col-sm-7 col-lg-6"
      >
        <h3
          class="pb-4"
        >
          Settings
        </h3>
         
        <div
          class="card p-4"
        >
          <h6
            class="pb-3"
          >
            Profile
          </h6>
           
          <a
            class="text-capitalize"
          >
            edit profile
          </a>
           
          <p
            class="text small"
          >
            Update your information
          </p>
           
          <a
            class="text-capitalize mt-3"
          >
            change password
          </a>
           
          <p
            class="text small"
          >
            Update your information
          </p>
           
          <a
            class="text-capitalize mt-3"
          >
            customize country
          </a>
           
          <p
            class="text small"
          >
            Update your information
          </p>
           
          <a
            class="text-capitalize mt-3"
          >
            customize category
          </a>
           
          <p
            class="text small"
          >
            Update your information
          </p>
           
          <a
            class="text-capitalize mt-3"
            data-toggle="modal"
            href="#feedback-modal"
          >
            give feedback
          </a>
           
          <p
            class="text small"
          >
            Update your information
          </p>
           
          <p
            class="text-capitalize font-weight-bold pt-4 auth"
          >
            authorization
          </p>
           
          <div
            class="pt-3 d-flex justify-content-between align-items-center"
          >
            <div
              class="pr-2"
            >
              <p
                class="text-capitalize"
              >
                newsletter subscription
              </p>
               
              <p
                class="text small"
              >
                Subscribe to receive article digest of your interest
              </p>
            </div>
             
            <div>
              <button
                aria-pressed="true"
                autocomplete="off"
                class="btn btn-sm btn-toggle active"
                data-toggle="button"
                type="button"
              >
                <div
                  class="square"
                />
              </button>
            </div>
          </div>
           
          <!---->
           
          <div
            class="pt-3 d-flex justify-content-between align-items-center"
          >
            <div>
              <p>
                Browser Push Notifications
              </p>
               
              <p
                class="text small"
              >
                All notifications
              </p>
            </div>
             
            <div>
              <button
                aria-pressed="true"
                autocomplete="off"
                class="btn btn-sm btn-toggle active"
                data-toggle="button"
                type="button"
              >
                <div
                  class="square"
                />
              </button>
            </div>
          </div>
           
          <!---->
           
          <div
            class="hr"
          />
           
          <!---->
           
          <div
            class="d-flex justify-content-between align-items-center"
          >
            <div>
              <p
                class="text-capitalize"
              >
                Become a Content Creator
              </p>
               
              <p
                class="text small mb-2"
              >
                Publish your article to the world
              </p>
               
              <a
                class="text-capitalize"
              >
                know more
              </a>
            </div>
             
            <div>
              <a
                class="btn btn-sm btn-primary text-capitalize"
              >
                apply
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <main
      id="bl-modal"
    >
      <div
        aria-hidden="true"
        aria-labelledby="customModal"
        class="modal fade"
        id="feedback-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header darkBackground"
            >
              <!---->
               
              <div>
                <h6>
                  <strong
                    class="alt-title"
                  >
                    Send Us Some Feedback
                  </strong>
                </h6>
                 
                <h6
                  class="mt-2 sub-title"
                >
                  Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!
                </h6>
              </div>
               
              <button
                aria-label="Close"
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  Ã—
                </span>
              </button>
            </div>
             
            <div
              class="modal-body"
            >
              <div
                class="bl-modal-content"
              >
                <p>
                  Enter your feedback here!
                </p>
                 
                <textarea
                  class="form-control mb-2 p-1"
                />
                 
                <label
                  class="ml-1 mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                   Bug
        
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                   Comment
        
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                   Other
        
                </label>
              </div>
            </div>
             
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-outline-primary"
                data-dismiss="modal"
                type="button"
              >
                Close
              </button>
               
              <button
                class="btn btn-primary"
                type="button"
              >
                Send Feedback
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</main>
`;
